### DISCUSSION TITLE
Error compiling for Arduino Nano RP2040 Connect board. RESOLVED!!

### DISCUSSION THREAD

**Discussion by ojkoIO** [JAndrassy/ArduinoOTA#146] [Q&A]
Hello,
I have a problem with the Arduino NanoRP2040 board.
I'm loading the OTASketchDownloadWifi sketch but when I launch the check I get this error:
```
/Users/ojko/Desktop/arduino/libraries/ArduinoOTA-master/src/utility/rp2_flash_boot.c:25:10: fatal error: RP2040.h: no files or directories of this type
   #include <RP2040.h> // CMSIS
            ^ ~~~~~~~~~
compilation finished.
More than one library found for "ArduinoHttpClient.h"
Used: / Users / ojko / Desktop / arduino / libraries / ArduinoHttpClient
Not used: / Users / ojko / Desktop / arduino / libraries / ArduinoHttpClient-master
exit status 1
Error compiling for Arduino Nano RP2040 Connect board.
```
```
/Users/ojko/Desktop/arduino/libraries/ArduinoOTA-master/src/utility/rp2_flash_boot.c:25:10: fatal error: RP2040.h: no files or directories of this type
   #include <RP2040.h> // CMSIS
            ^ ~~~~~~~~~
compilation finished.
More than one library found for "ArduinoHttpClient.h"
Used: / Users / ojko / Desktop / arduino / libraries / ArduinoHttpClient
Not used: / Users / ojko / Desktop / arduino / libraries / ArduinoHttpClient-master
exit status 1
Error compiling for Arduino Nano RP2040 Connect board.
```

**Comment by JAndrassy** [comment]
with [Pico core](https://github.com/earlephilhower/arduino-pico)?

**Comment by ojkoIO** [comment]
Il giorno gio 31 mar 2022 alle 18:02 Juraj Andrássy <
***@***.***> ha scritto:

> with Pico core <https://github.com/earlephilhower/arduino-pico>?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jandrassy/ArduinoOTA/discussions/146#discussioncomment-2479566>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXYZKGNZIN2LKLOQ3UXVC7DVCXEAPANCNFSM5SFVT25A>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
No, with Arduino IDE

**Comment by JAndrassy** [comment]
the ArduinoOTA library supports RP2040 only with the Pico core, not with the Arduino mbed core

**Reply by ojkoIO** [reply]
ok, thanks tonight I install the pico core and try.

Thanks.

**Reply by ojkoIO** [reply]
Hi, I have installed PICO CORE on the arduino IDE and everything is loaded on the RP2040 nano board.

I am trying to use OTA to automatically load the firmware updated according to the version from the server.
As an advanced example: OTASketchDonwnloadWifi.

But in the code I look for a .bin file and if I compile the sketch I end up with a .uf2 file


`Created for ArduinoOTA library in December 2020
  by Nicola Elia
  based on Juraj Andrassy sample sketch 'OTASketchDownload'
*/

#include <ArduinoOTA.h> // only for InternalStorage
#include <WiFiNINA.h>
#include <ArduinoHttpClient.h>

// Please enter your WiFi sensitive data in the arduino_secrets.h file
#include "arduino_secrets.h"

const short VERSION = 2;

const char MY_SSID[] = SECRET_SSID; // Loaded from arduino_secrets.h
const char MY_PASS[] = SECRET_PASS; // Loaded from arduino_secrets.h

 WiFiClient    wifiClient;  // HTTP
//WiFiSSLClient wifiClientSSL;  // HTTPS
int status = WL_IDLE_STATUS;

void handleSketchDownload() {
  const char* SERVER = "xxxxxxx";  // Set your correct hostname
  const unsigned short SERVER_PORT = 80;     // Commonly 80 (HTTP) | 443 (HTTPS)
  const char* PATH = "/update-v%d.bin";       // Set the URI to the .bin firmware
  const unsigned long CHECK_INTERVAL = 6000;  // Time interval between update checks (ms)

  // Time interval check
  static unsigned long previousMillis;`

can you help me.

Thank you.

**Comment by JAndrassy** [comment]
and did you read the README?

>... platform.local.txt from extras folder has to be copied into boards package installation folder.

**Reply by ojkoIO** [reply]
I copied the platform.local.txt file to the folder:

/ Users / ojko / Arduino15 / packages / rp2040 / hardware / rp2040 / 1.13.1

but nothing finds the file on the server but then it is as if it crashes.

file extension to be searched on the server has been changed from .bin to .ino.ard.uf2

the serial tells me that:

12: 11: 19.009 -> Check for update file /update-v3.ino.ard.uf2
12: 11: 20.051 -> Update status code: 404
12: 11: 39.023 -> Check for update file /update-v3.ino.ard.uf2
12: 11: 40.050 -> Update status code: 200
12: 11: 40.050 -> Server returned update file of size 152576 bytes
12: 11: 42.221 -> Sketch update apply and reset.

but then everything stops.

**Reply by JAndrassy** [reply]
bin must be used for OTA update. use "Export compiled binary" to get the bin into sketch folder

**Reply by ojkoIO** [reply]
Hi, thanks solved now it works.

I don't know why but the arduino IDE version 1.8.19 does not generate the correct .bin file for me, tried with the arduino IDE 2.0 version and everything works perfectly.

Thanks again.
